### ISSUE TITLE
Import-Module -Prefix '' on a module with a Default Prefix defined does not remove the prefix

### ISSUE THREAD

**Issue by JustinGrote** [PowerShell/PowerShell#16936]
### Prerequisites

- [X] Write a descriptive title.
- [X] Make sure you are able to repro it on the [latest released version](https://github.com/PowerShell/PowerShell/releases)
- [X] Search the existing issues.
- [X] Refer to the [FAQ](https://github.com/PowerShell/PowerShell/blob/master/docs/FAQ.md).
- [X] Refer to [Differences between Windows PowerShell 5.1 and PowerShell](https://docs.microsoft.com/powershell/scripting/whats-new/differences-from-windows-powershell).

### Steps to reproduce

1. Make a module, and set Prefix to 'Test'
2. `Import-Module MyTestModule -Prefix ''`
3. All commands will still have test prefix

This is because the prefix test checks for IsNullOrEmpty instead of just null.
```
            bool usePrefix = !string.IsNullOrEmpty(prefix);
```

As such it is impossible to override the prefix which affects the ability to write pester tests on modules, etc. and use the original function names so that things like references and go to definition work in IDEs like vscode.

### Workaround
Make a secondary manifest that references the same module but does not have the prefix specified.

### Expected behavior

```console
Commands without prefix
```

### Actual behavior

```console
Commands have default prefix
```

### Environment data

```powershell
PS 7.2.1
```

**Comment by iSazonov** [comment]
As alternative for discussion - introduce new parameter (-NoPrefix/-RemovePrefix).

Also question for discussion - is the prefix a strong contact from module author?

**Comment by JustinGrote** [comment]
@iSazonov I don't think it should be a strong contract. A "proper" module (at least script wise) I write as if it has no prefix, all the commands and references are of the non-prefixed version. If a prefix is specified, everything continues to work as expected. The user can specify multiple prefixes potentially and import the module multiple times to address multiple target scenarios, so they can use `Get-SourceMyFunction` and `Get-DestinationMyFunction` for instance if the connection context is kept at module scope.

I do think that the specified bool should not be IsNullOrEmpty but simply a null check, so that an empty '' can indicate `I don't want a prefix`, however I understand that is more breaking than `-NoPrefix` so I"m fine with that too.

**Comment by JustinGrote** [comment]
I made the following workaround, the regexes are fragile but it works for my use case
```powershell
function Import-ModuleWithoutPrefix {
  [CmdletBinding()]
  param(
    [Parameter(Mandatory, ValueFromPipeline)]$ManifestPath,
    [Switch]$Force,
    $TempPath = 'TEMP:'
  )
  $path = Resolve-Path $ManifestPath
  $manifestDir = Split-Path $Path -Parent
  $fileName = [io.path]::GetFileNameWithoutExtension($path)
  #Manifest file name must be the same
  $tempPath = "TEMP:\$(New-Guid)\$fileName.psd1"

  $manifestContent = Get-Content -Raw $path
  $rootModuleRegex = '(?<=RootModule\s*?=\s*?[''"])(.+?)(?=[''"])'
  if ($manifestContent -notmatch $rootModuleRegex) {
    throw "Could not find RootModule entry in manifest $path"
  }
  $rootModulePath = [io.path]::IsPathRooted($matches[1]) ? $matches[1] : $(Join-Path $manifestDir $matches[1])
  $resolvedRootModulePath = Resolve-Path $rootModulePath
  $manifestContent = $manifestContent -replace $rootModuleRegex, $resolvedRootModulePath

  #Remove any prefix entry present. If this doesn't exist we don't error out
  $manifestContent = $manifestContent -replace 'DefaultCommandPrefix\s*?=\s*?[''"].+?[''"]'

  #This should be unique
  New-Item -ItemType File -Force -Path $tempPath -Value $manifestContent | Out-Null

  Write-Verbose "Importing module $path without prefix from $tempPath"
  Import-Module $tempPath -Force:$Force -PassThru
}
```

**Comment by iSazonov** [comment]
We rolled back PR with fix due to a regression. The code is not trivial and has low test coverage.
We need to add more tests for importing modules with/without (default) prefixes.
Prefix parameter is internally BasePrefix which is initialized with `string.Empty`. So we should change this to null. But BasePrefix is used in some code branches which should be reviewed and fixed with covering by tests.

**Comment by JustinGrote** [comment]
@iSazonov thanks for the investigation!

**Comment by JustinGrote** [comment]
Still Active
